src/main/java/edu/cornell/kfs/fp/batch/service/impl/AuxiliaryVoucherDocumentGenerator.java

```diff
@@ -0,0 +1,177 @@
+        return AuxiliaryVoucherDocument.class;
+    }
+
+    @Override
+    protected <A extends AccountingLine> A buildAccountingLine(
+            Class<A> accountingLineClass, String documentNumber, AccountingXmlDocumentAccountingLine xmlLine) {
+        A accountingLine = super.buildAccountingLine(accountingLineClass, documentNumber, xmlLine);
+        validateAndPopulateAccountingLineWithDebitCreditAmounts(accountingLine, xmlLine);
+        return accountingLine;
+    }
+
+    protected <A extends AccountingLine> void validateAndPopulateAccountingLineWithDebitCreditAmounts(
+            A accountingLine, AccountingXmlDocumentAccountingLine xmlLine) {
+        if (hasAmountBeenSpecified(xmlLine.getAmount())) {
+            throw buildNewValidationException(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_AV_INVALID_AMOUNT_TYPE);
+        }
+        
+        KualiDecimal debitAmount = xmlLine.getDebitAmount();
+        KualiDecimal creditAmount = xmlLine.getCreditAmount();
+        if (hasAmountBeenSpecified(debitAmount)) {
+            if (hasAmountBeenSpecified(creditAmount)) {
+                throw buildNewValidationException(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_AV_DEBIT_AND_CREDIT_AMOUNT);
+            }
+            accountingLine.setDebitCreditCode(KFSConstants.GL_DEBIT_CODE);
+            accountingLine.setAmount(debitAmount);
+        } else if (hasAmountBeenSpecified(creditAmount)) {
+            accountingLine.setDebitCreditCode(KFSConstants.GL_CREDIT_CODE);
+            accountingLine.setAmount(creditAmount);
+        } else {
+            throw buildNewValidationException(CuFPKeyConstants.ERROR_CREATE_ACCOUNTING_DOCUMENT_AV_AMOUNT_REQUIRED);
+        }
+    }
+
+    protected boolean hasAmountBeenSpecified(KualiDecimal amount) {
+        return amount != null && !KualiDecimal.ZERO.equals(amount);
+    }
+
+    @Override
+    protected void populateCustomAccountingDocumentData(AuxiliaryVoucherDocument document, AccountingXmlDocumentEntry documentEntry) {
+        DateTime documentCreateDate = getDocumentCreateDate(document);
+        Date documentCreateDateWithoutTime = new Date(documentCreateDate.getMillis());
+        
+        AccountingPeriod period = findEligibleAccountingPeriod(document, documentEntry);
+        document.setAccountingPeriod(period);
+        
+        String avTypeCode = validateAndGetAuxiliaryVoucherType(documentEntry);
+        document.setTypeCode(avTypeCode);
+        
+        Optional<Date> reversalDate = validateAndGetReversalDateIfApplicable(
+                documentEntry, period, avTypeCode, documentCreateDateWithoutTime);
+        if (reversalDate.isPresent()) {
+            document.setReversalDate(reversalDate.get());
+        }
+    }
+
+    protected DateTime getDocumentCreateDate(AuxiliaryVoucherDocument document) {
+        WorkflowDocument workflowDocument = document.getDocumentHeader().getWorkflowDocument();
+        return workflowDocument.getDateCreated();
```

Comment:
Why do we need to get the date created from the workflow document?  Wouldn't it always be the current date?

Reply:
Good point, we don't run this job at around midnight so it should be safe to assume that the current date is also the create date. I'll adjust this accordingly.